onnxruntime/contrib_ops/cuda/bert/attention_impl.cu

```diff
@@ -54,84 +54,115 @@ size_t GetAttentionWorkspaceSize(size_t element_size, int batch_size, int num_he
 }
 
 template <typename T, unsigned TPB>
-__device__ inline void Softmax(const int ld, const int last_valid, const T* input, T* output) {
+__device__ inline void Softmax(const int ld, const int num_valid, const T* input, T* output) {
   using BlockReduce = cub::BlockReduce<float, TPB>;
   __shared__ typename BlockReduce::TempStorage tmp_storage;
 
-  __shared__ float reverse_z;
+  __shared__ float sum_reverse_block;
+  __shared__ float max_block;
 
   float thread_data(0);
```

Comment:
>thread_data [](start = 8, length = 11)

If all inputs are negative, 0 is not a good init value, should be something like -1e38

Reply:
If all inputs are negative, e^x won't have instability issue, thus it is good to subtract 0 (max) from the value list.